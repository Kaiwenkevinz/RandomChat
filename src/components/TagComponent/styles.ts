import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#007AFF', // 蓝色背景
    borderRadius: 4, // 圆角边框
    paddingHorizontal: 8, // 左右留白
    paddingVertical: 4, // 上下留白
    marginHorizontal: 4, // 左右留白
  },
  text: {
    color: '#FFFFFF', // 白色文本
    textAlign: 'center',
    fontSize: 14, // 字体大小
    fontWeight: 'bold', // 粗体字
  },
});
