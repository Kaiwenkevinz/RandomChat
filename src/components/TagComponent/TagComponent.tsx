import {View, Text} from 'react-native';
import React from 'react';
import {styles} from './styles';

interface TagProps {
  text: string;
  backgroundColor?: string;
  textColor?: string;
}

const TagComponent = ({text, backgroundColor, textColor}: TagProps) => {
  return (
    <View
      style={[
        styles.container,
        backgroundColor ? {backgroundColor: backgroundColor} : {},
      ]}>
      <Text style={[styles.text, textColor ? {color: textColor} : {}]}>
        {text}
      </Text>
    </View>
  );
};

export default TagComponent;
