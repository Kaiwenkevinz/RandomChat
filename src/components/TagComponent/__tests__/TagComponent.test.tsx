import renderer from 'react-test-renderer';
import React from 'react';
import TagComponent from '../TagComponent';

it('renders correctly', () => {
  const tree = renderer.create(<TagComponent text="tag text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
