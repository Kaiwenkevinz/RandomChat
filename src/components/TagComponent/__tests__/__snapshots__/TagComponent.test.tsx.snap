// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#007AFF",
        "borderRadius": 4,
        "marginHorizontal": 4,
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {},
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        {},
      ]
    }
  >
    tag text
  </Text>
</View>
`;
